src/App.js

```diff
@@ -1,61 +1,52 @@
-import React, {
-  useState
-} from 'react';
+import React, { useState } from 'react';
 import WeatherBox from './components/WeatherBox';
 import Attractions from './components/Attractions';
-
-
-import {
-  OPENWEATHER_API_KEY,
-  GRAPH_HOPPER_API_KEY,
-  OPENTRIP_MAP_API_KEY
-} from './config/keys';
-
-const weatherApi = {
-  key: OPENWEATHER_API_KEY,
-  base: "https://api.openweathermap.org/data/2.5/"
-}
-
-const geotagApi = {
-  key: GRAPH_HOPPER_API_KEY,
-  base: "https://graphhopper.com/api/1/"
-}
-
-const placesApi = {
-  key: OPENTRIP_MAP_API_KEY,
-  base: "https://api.opentripmap.com/0.1/en/"
-}
+import { openWeatherApi } from './apiHandling/openWeatherApi';
+import { geotagApi } from './apiHandling/geotagApi';
+import { openTripApi } from './apiHandling/openTripApi';
 
 function App() {
 
   const [query, setQuery] = useState('');
   const [weather, setWeather] = useState({});
   const [attractions, setAttractions] = useState({});
   const [coords, setCoords] = useState({});
-  const [placeDesc, setDesc] = useState({});
+  const [placeDesc, setDesc] = useState([]);
+  const [places, setPlaces] = useState([]);
+  const [maxShown, setMaxShown] = useState(10);
+
+  let loadAtOnce = 10;
+
+  const loadMore = () => {
+    setMaxShown(prevMaxShown => prevMaxShown + 10);
+    loadDescriptions(places);
+  }
 
   const search = evt => {
       if (evt.key === "Enter") {
           setCoords({});
-
-          loadWeather();
-          let coordinates = loadCoordinades();
-          loadAttractions(coordinates);
+          setAttractions({});
+          setMaxShown(10);
+          if (query.length > 0){
+            loadWeather();
+            let coordinates = loadCoordinades();
+            coordinates.then( point => loadAttractions(point));
+          }   
       }
   }
 
   function loadWeather() {
-      fetch(`${weatherApi.base}weather?q=${query}&units=metric&APPID=${weatherApi.key}`)
-          .then(res => res.json())
-          .then(result => {
-              setWeather(result);
-          });
+      openWeatherApi.fetchOpenWeatherApi(query)
+        .then(res => res.json())
+        .then(result => {
+            setWeather(result);
+        });
   }
 
   function loadCoordinades() {
```

Comment:
coordinates

Reply:
if life gives you melons you may be dyslexic meaning